clio-dataaccess-model/src/main/scala/org/broadinstitute/clio/server/dataaccess/elasticsearch/ElasticsearchIndex.scala

```diff
@@ -24,11 +23,24 @@ import org.broadinstitute.clio.util.model.{DataType, UpsertId}
 class ElasticsearchIndex[+CI <: ClioIndex](
   val clioIndex: CI,
   final val indexName: String,
+  final val orderedKeyFields: Seq[String],
   private[elasticsearch] val fieldMapper: ElasticsearchFieldMapper,
   private[elasticsearch] val defaultFields: Json = Json.obj()
 ) extends ModelAutoDerivation {
   import clioIndex.implicits._
 
+  private val keyFields = FieldMapper[clioIndex.KeyType].fields.map {
+    case (k, v) => ElasticsearchUtil.toElasticsearchName(k) -> v
+  }
+  private val metadataFields = FieldMapper[clioIndex.MetadataType].fields.map {
+    case (k, v) => ElasticsearchUtil.toElasticsearchName(k) -> v
+  }
+
+  assert(
```

Comment:
I don't think we should have asserts in our production code. Doing it in a test, for each index, should be sufficient.

Reply:
I worry about having no guards here, where before we at least had one source of truth (the type declaration). Tests will work until we add the next index and forget about to add the very-important-unit-test.

If I changed it to `require` instead of `assert`, would that be better? It throws an `IllegalArgumentException` instead of an `AssertionError`.